### PR TITLE
Turn off leak tracker in master to make found leaks not blocking.

### PR DESCRIPTION
**Pull request by polina-c**
See discussion: https://discord.com/channels/608014603317936148/608021234516754444/1174820774712975480

### CODE REVIEW

**Comment by matanlurey** [comment]
I don't work on the framework so I'm the wrong person to review this PR, reviewed myself. Thanks!

File: packages/flutter/test/flutter_test_config.dart [hunk 1]
```diff
@@ -27,9 +27,9 @@ Future<void> testExecutable(FutureOr<void> Function() testMain) {
 
   // TODO(polina-c): clean up leaks and stop ignoring them.
   // https://github.com/flutter/flutter/issues/137311
+  // Leak tracking is off by default. To temporary enable it add `.withTrackedAll()` after `settings`.
   LeakTesting.settings = LeakTesting
     .settings
```
Comment (jacob314) [inline packages/flutter/test/flutter_test_config.dart:32]: This is fine to unblock for now but you might want to consider using an environment variable to allow users to turn on leak tracking by setting the environment variable when calling `flutter test`.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/flutter/test/flutter_test_config.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:32]: Is there difference in effort to add flag and to add this line of code?

Created PR that will show memory leaks: https://github.com/flutter/flutter/pull/138581
The plan is to refresh it manually and check if there are leaks.

Do you happen to have other automation in mind?

**Pull request merged**